test/e2e/clusterctl_upgrade_test.go

```diff
@@ -51,17 +49,6 @@ var _ = Describe("When testing clusterctl upgrades (v1.0=>current)", func() {
 			WorkloadKubernetesVersion: "v1.23.17",
 			MgmtFlavor:                "topology",
 			WorkloadFlavor:            "",
-			// This check ensures that ownerReference apiVersions are updated for all types after the upgrade.
-			PostUpgrade: func(proxy framework.ClusterProxy, namespace, clusterName string) {
```

Comment:
I don't think these tests make sense any more as there's no longer any APIVersion upgrades in our clusterctl_upgrade tests. 

This is something we should make sure is part of the introduction of `v1beta1` or whatever the next APIVersion is.

Reply:
+1 except we already have v1beta1 :D

Reply:
Do you think we still need it now that we have ValidateOwnerReferencesOnUpdate?